packages/components/badge/src/Badge/Badge.tsx

```diff
@@ -5,49 +5,83 @@ import {
   type CommonProps,
   type PropsWithHTMLElement,
   type ExpandProps,
+  type PickUnion,
 } from '@contentful/f36-core';
 
 import type { BadgeSize, BadgeVariant } from '../types';
 import { getBadgeStyles } from './Badge.styles';
 
-export interface BadgeInternalProps extends CommonProps {
-  /**
-   * Sets the size of the component
-   * @default default
-   */
-  size?: BadgeSize;
-  /**
-   * Determines the variation of the component
-   * @default primary
-   */
-  variant?: BadgeVariant;
-
-  children: React.ReactNode;
-}
+type BadgeSizeWithIconProps =
+  | {
+      /**
+       * Sets the size of the component
+       * @default default
+       */
+      size?: PickUnion<BadgeSize, Exclude<BadgeSize, 'small'>>;
```

Comment:
If I got the logic correctly, we can simplify this:
```suggestion
      size?: Exclude<BadgeSize, 'small'>;
```